demisto_sdk/commands/common/hook_validations/deprecation.py

```diff
@@ -0,0 +1,193 @@
+        """
+        for script in self.script_section:
+            for script_val in script.values():
+                if script_val.get("deprecated"):
+                    continue
+                depends_commads_list = script_val.get("depends_on")
+                if depends_commads_list:
+                    for command in depends_commads_list:
+                        if command in deprecated_commands_list:
+                            script_path = script_val.get("file_path", "")
+                            if command in usage_dict and script_path not in usage_dict.get(command, []):
+                                usage_dict.get(command, []).append(script_path)
+                            elif command not in usage_dict:
+                                usage_dict[command] = [script_path]
+        return usage_dict
+
+    def collect_non_deprecated_scripts_implemeting_another_script(self, script_name: str, usage_list: List[str]):
+        """
+        List all the paths of scripts that are using the given script and are non-deprecated.
+
+        Args:
+            script_name (str): The name of the script that is currently being checked.
+            usage_list (list): A list of all the files that use the given script to update the found scripts into.
+        """
+        for script in self.script_section:
+            for script_val in script.values():
+                if script_val.get("deprecated"):
+                    continue
+                depends_commads_list = script_val.get("depends_on")
+                if depends_commads_list and script_name in depends_commads_list:
+                    usage_list.append(script_val.get("file_path"))
+        return usage_list
+
+    def collect_non_deprecated_playbooks_that_use_a_given_playbook_or_script(self, curent_entity_name: str, usage_list: List[str], key_to_check: str):
+        """
+        Collect from id_set all the playbooks that are non-deprecated and are implementing the given entity (script/playbook).
+
+        Args:
+            curent_entity_name (str): The name of the script / playbook currently being checked.
+            usage_list (list): A list of all the file paths that are using the script / playbook currently being checked.
+            key_to_check (str): The field in which the right entity to check is located at.
+        """
+        for playbook in self.playbook_section:
+            for playbook_val in playbook.values():
+                if playbook_val.get('deprecated'):
+                    continue
+                implementing_entities = playbook_val.get(key_to_check)
+                if implementing_entities:
+                    usage_list = self.validate_playbook_or_script_not_in_playbook(usage_list, curent_entity_name, implementing_entities, playbook_val)
+        return usage_list
+
+    def validate_playbook_or_script_not_in_playbook(self, usage_list: List[str], curent_entity_name: str, implementing_entities: List[str], playbook: Dict):
```

Comment:
this method does not seem to validate, please rename it to `find_implementing_playbooks_and_scripts`